docs/src/adr/019adr-changelog-management.md

```diff
@@ -0,0 +1,182 @@
+for automatically generating changelogs. These tools extract changelogs from
+various different sources such as commit messages, issues, or pull requests.
+    
+#### Pros
+
+- Requires no new development
+- Would eliminate the need to add changelog entries in a separate file
+
+#### Cons
+
+- Developers have to learn the annotation format required by the tool, and they
+  have to remember to deploy it in their commits/PRs/issues.
+- Infrastructure maintainers have to learn how to configure and maintain the tool.
+- We pickup a dependency external to both the company and our project
+- Changelogs are meant to communicate a very specific kind of information to a
+  specific audience. Git commit messages, pull requests, and issues are for
+  communicating quite different information to an entirely different audience.
+  Requiring that these communication channels be overloaded risks degrading the
+  quality of communications in all the conflated channels.
+  - However, it's possible we could work out some conventions to add empty
+    commits or designate special PR labels to work around this interference?
+
+
+### 2. Use Unclog
+
+@thanethompson has written the [unclog][] utility, which is used in both
+[informalsystems/tendermint-rs][] and [informalsystems/ibc-rs][]. The tool
+stores changelog entries in a directory structure, with each entry in its own
+file, ensuring merge conflicts can't arise. 
+
+Entries are added via the CLI. E.g.:
+
+```sh
+unclog add --id some-new-feature \
+  --issue 23 \
+  --section breaking-changes \
+  --message "Some *new* feature"
+```
+
+If the `--messages` argument is omitted, it will open your configured editor for
+authoring the message.
+
+The tool has subcommands to generate and update the `CHANGES.md`, and it supports
+a variety of options via its TOML configuration.
+
+[unclog]: https://github.com/informalsystems/unclog
+[informalsystems/tendermint-rs]: https://github.com/informalsystems/tendermint-rs
+[informalsystems/ibc-rs]: https://github.com/informalsystems/ibc-rs
+
+#### Pros
+
+- Requires no new development
+- It is developed internally to informal
+
+#### Cons
+
+- Adds cognitive and procedural overhead to adding changelog entries
+- Would add a rust dependency to our devenv
+- Requires external contributors to either install a CLI tool or figure out a
+  relatively complex file and folder structure to add changelog entries
+
+### 3. Use a custom git merge driver
+
+If we wrote a custom merge driver to work on the simple changelog format we
+could continue our current practice, and just fall back to the custom merge
+driver in case of merge conflicts.
+
+#### Pros
+
+- Allows us to keep our current process of simply updating a markdown file,
+  with which, moreover, most devs are already familiar
+
+#### Cons
+
+- Could require external dependencies to be installed, depending on
+  implementation.
+- It's not possible to configure custom merge drivers for github, so we'd either
```

Comment:
Did you test this? libgit2 which seems to drive the merge on github [seems to support this](https://github.com/libgit2/libgit2/pull/3564#issuecomment-217581428).

Reply:
Right, there's no problem to install custom merge drivers on github runners in CI. But github doesn't allow configuration of custom merge drivers on the auto merge functionality it uses to update PRs. And without that, we'll have the added overhead of writing a bot/action that monitors our pending PRs and then updates them with the custom merge strategy installed.

Otherwise we're just back to devs having to fix these locally. While it would save a bit of time having a custom merge driver for these files rather than having to do it manually, it doesn't really clear up the flow the way we want.

That said, I really do like this solution, and I prototyped a very rough merge driver for this last weekend for fun. I think I may pursue that as a side project regardless. Personally, I still am inclined to think that keeping changes in a single markdown file is the nicest way to record them :D

Reply:
> Without that functionality, we'll have the added overhead of writing a bot/action that monitors our pending PRs and then updates them with the custom merge strategy installed.

Yeah no, that seems too complicated...

> Personally, I still am inclined to think that keeping changes in a single markdown file is the nicest way to record them

I agree!